bentoml/_internal/server/grpc/servicer.py

```diff
@@ -0,0 +1,189 @@
+
+        async def Call(  # type: ignore (no async types) # pylint: disable=invalid-overridden-method
+            self,
+            request: pb.Request,
+            context: BentoServicerContext,
+        ) -> pb.Response | None:
+            if request.api_name not in service.apis:
+                raise UnprocessableEntity(
+                    f"given 'api_name' is not defined in {service.name}",
+                ) from None
+
+            api = service.apis[request.api_name]
+            response = pb.Response()
+
+            try:
+                _use_internal_bytes_contents = False
+                # map by default is optional ,and hence we need to check
+                # whether multipart is an empty dict or not.
+                if len(request.multipart) != 0:
+                    io_proto = request.multipart
+                elif request.HasField(api.input.accepted_proto_fields):
+                    io_proto = getattr(request, api.input.accepted_proto_fields)
+                elif request.HasField("_internal_bytes_contents"):
+                    io_proto = request._internal_bytes_contents
+                    _use_internal_bytes_contents = True
+                else:
+                    raise InvalidArgument(
+                        f"Neither '{api.input.accepted_proto_fields}' or '_internal_bytes_contents' field is found in the request message.",
+                    ) from None
+                input_ = await api.input.from_proto(
+                    io_proto, _use_internal_bytes_contents=_use_internal_bytes_contents
+                )
+                if asyncio.iscoroutinefunction(api.func):
+                    if isinstance(api.input, Multipart):
+                        output = await api.func(**input_)
+                    else:
+                        output = await api.func(input_)
+                else:
+                    if isinstance(api.input, Multipart):
+                        output = await anyio.to_thread.run_sync(api.func, **input_)
+                    else:
+                        output = await anyio.to_thread.run_sync(api.func, input_)
+                protos = await api.output.to_proto(output)
+                response = pb.Response(**{api.output.accepted_proto_fields: protos})
+            except BentoMLException as e:
+                log_exception(request, sys.exc_info())
+                await context.abort(code=grpc_status_code(e), details=e.message)
+            except (RuntimeError, TypeError, NotImplementedError):
+                log_exception(request, sys.exc_info())
+                await context.abort(
+                    code=grpc.StatusCode.INTERNAL,
+                    details="A runtime error has occurred, see stacktrace from logs.",
+                )
+            except Exception:  # pylint: disable=broad-except
+                log_exception(request, sys.exc_info())
+                await context.abort(
+                    code=grpc.StatusCode.UNKNOWN,
```

Comment:
It's still a server error, so we should probably return this.

```suggestion
                    code=grpc.StatusCode.INTERNAL,
```